src/api/app/models/staging/workflow.rb

```diff
@@ -108,4 +109,15 @@ def add_reviewer_group
     project.relationships.find_or_create_by(group: managers_group, role: role)
     project.store
   end
+
+  def update_reviewer_group
+    return unless changes[:managers_group_id]
+
+    old_managers_group = Group.find(changes[:managers_group_id].first)
+    new_managers_group = managers_group
+
+    role = Role.find_by_title('reviewer')
+    relationship = project.relationships.find_by(group: old_managers_group, role: role)
+    relationship.update(group: new_managers_group)
+  end
 end
```

Comment:
This really should be one callback. There is already a before_update hook looking at the group. Basically what needs to be done on group update:

- Change the reviewer role in target project (make sure the project is stored to backend afterwards, I'm not sure this happens here)
- Change the maintainer role in all staging projects (happens in the other callback, but I'm not sure it stores the project either)
- Change all requests in backlog by adding review by_group: new and accepting review by_group: old (in that order)

Reply:
shouldn't we pack these "3 steps" in one transaction?

Reply:
Note however that rollback needs special handling of the backend